sources/procfs.go

```diff
@@ -409,40 +333,70 @@ func (s *lustreProcfsSource) Update(ch chan<- prometheus.Metric) (err error) {
+		{pattern: "create", index: 1},
+		{pattern: "get_info", index: 1},
+		{pattern: "set_info_async", index: 1},
+		{pattern: "connect", index: 1},
+		{pattern: "ping", index: 1},
+	}
+	for _, operation := range operationSlice {
+		opStat := regexCaptureString(operation.pattern+" .*", statsFile)
+		if len(opStat) < 1 {
+			continue
+		}
+		r, err := regexp.Compile(" +")
+		if err != nil {
+			continue
+		}
+		bytesSplit := r.Split(opStat, -1)
+		result, err := strconv.ParseUint(bytesSplit[operation.index], 10, 64)
+		if err != nil {
+			return nil, err
+		}
+		l := lustreStatsMetric{
+			title:           promName,
+			help:            helpText,
+			value:           result,
+			extraLabel:      "operation",
+			extraLabelValue: operation.pattern,
+		}
+		metricList = append(metricList, l)
+	}
+	return metricList, nil
+}
+
+func getStatsIOMetrics(statsFile string, helpText string, promName string) (metricList []lustreStatsMetric, err error) {
 	// bytesSplit is in the following format:
 	// bytesString: {name} {number of samples} 'samples' [{units}] {minimum} {maximum} {sum}
 	// bytesSplit:   [0]    [1]                 [2]       [3]       [4]       [5]       [6]
 	bytesMap := map[string]multistatParsingStruct{
-		readSamplesHelp:     {pattern: "read_bytes .*", index: 1},
-		readMinimumHelp:     {pattern: "read_bytes .*", index: 4},
-		readMaximumHelp:     {pattern: "read_bytes .*", index: 5},
-		readTotalHelp:       {pattern: "read_bytes .*", index: 6},
-		writeSamplesHelp:    {pattern: "write_bytes .*", index: 1},
-		writeMinimumHelp:    {pattern: "write_bytes .*", index: 4},
-		writeMaximumHelp:    {pattern: "write_bytes .*", index: 5},
-		writeTotalHelp:      {pattern: "write_bytes .*", index: 6},
-		openHelp:            {pattern: "open .*", index: 1},
-		closeHelp:           {pattern: "close .*", index: 1},
-		getattrHelp:         {pattern: "getattr .*", index: 1},
-		setattrHelp:         {pattern: "setattr .*", index: 1},
-		getxattrHelp:        {pattern: "getxattr .*", index: 1},
-		setxattrHelp:        {pattern: "setxattr .*", index: 1},
-		statfsHelp:          {pattern: "statfs .*", index: 1},
-		seekHelp:            {pattern: "seek .*", index: 1},
-		readdirHelp:         {pattern: "readdir .*", index: 1},
-		truncateHelp:        {pattern: "truncate .*", index: 1},
-		allocInodeHelp:      {pattern: "alloc_inode .*", index: 1},
-		removexattrHelp:     {pattern: "removexattr .*", index: 1},
-		unlinkHelp:          {pattern: "unlink .*", index: 1},
-		inodePermissionHelp: {pattern: "inode_permission .*", index: 1},
-		createHelp:          {pattern: "create .*", index: 1},
-		getInfoHelp:         {pattern: "get_info .*", index: 1},
-		setInfoAsyncHelp:    {pattern: "set_info_async .*", index: 1},
-		connectHelp:         {pattern: "connect .*", index: 1},
-		pingHelp:            {pattern: "ping .*", index: 1},
-	}
-	bytesString := regexCaptureString(bytesMap[helpText].pattern, statsFile)
+		readSamplesHelp:  {pattern: "read_bytes .*", index: 1},
```

Comment:
Also, along the lines of #88, it might be a good idea to transition to the number regex pattern we use in all other functions as opposed to matching strings, both for consistency and safety. Though, this probably belongs in another PR which I can certainly pickup in the future.